### PR TITLE
[Feature] Configuration based on environment variable

### PR DESCRIPTION
**Pull request by ZhiHanZ**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary
Support to finalize configuration through environment variable instead of toml file

## Changelog

- New Feature

## Related Issues

fixes https://github.com/datafuselabs/datafuse/issues/907

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: fusequery/query/src/configs/config.rs [hunk 1]
```diff
@@ -31,87 +31,76 @@ lazy_static! {
...
-    #[structopt(long, env = "STORE_API_PASSWORD", default_value = "root")]
+    #[structopt(long, default_value = "root")]
     pub store_api_password: String,
 
-    #[structopt(long, short = "c", env = "CONFIG_FILE", default_value = "")]
+    #[structopt(long, short = "c", default_value = "")]
     pub config_file: String,
 }
 
+// adaptor for environment variable
```
Comment (BohuTANG) [inline fusequery/query/src/configs/config.rs:83]: Hello,
Why not based on the `Config`?
In this case, we need to maintain two configurations

**Comment by sundy-li** [comment]
Why not introduce a new field named `config: String` in `Config`

1. If the config is specified, then call load_from_config to build a config, this config can merge config parsed from env.
2. If the config is not specified, keep it as original behavior, just parse the config from env.

**Comment by ZhiHanZ** [comment]
> Why not introduce a new field named `config: String` in `Config`
> 
> 1. If the config is specified, then call load_from_config_file to build a config, this config can merge config parsed from env.
> 2. If the config is not specified, keep it as original behavior, just parse the config from env.

I think config_file field already provided to us, and this implementation allow Configuration to be modified through environment variable after it has already loaded a config file through command line arguments

**Comment by BohuTANG** [comment]
@sundy-li , @ZhiHanZ requirement is that don't let the toml file override the env, this is useful for such k8s configed by environment variable.

This looks ok to me.

File: fusequery/query/src/bin/fuse-query.rs [hunk 2]
```diff
@@ -31,6 +31,8 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         conf = Config::load_from_toml(conf.config_file.as_str())?;
     }
```
Comment (BohuTANG) [inline fusequery/query/src/bin/fuse-query.rs:33]: It's better to leave a comments here why  `load_from_env` again

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**